src/bootstrap/src/ferrocene/dist.rs

```diff
@@ -431,3 +405,39 @@ impl Step for GenerateBuildMetadata {
         builder.copy_to_folder("ferrocene/packages.toml".as_ref(), dist_dir.as_ref());
     }
 }
+
+impl GenerateBuildMetadata {
+    pub(super) fn channel(
+        rust_channel: &str,
```

Comment:
This can be accessed with `builder.config.channel`. I'd accept a builder instance instead of this argument.

Reply:
Yeah that is the expected value though, locally you're using the `dev` channel. `src/ci/channel` is never read by bootstrap, it is configured by CI [by setting the `rust.channel` key in `config.toml` to the value of that file](https://github.com/ferrocene/ferrocene/blob/bc59de8a10dacc2b649df88f521cb51ecbcf58d9/ferrocene/ci/configure.sh#L233-L240), and upstream CI does the same.

If we were to read `src/ci/channel` ourselves we would stop respecting `rust.channel`, which would be super confusing. I'd prefer if we'd access it and then treat `dev` as `nightly` in the match above.